rxjava/src/test/java/com/linecorp/armeria/common/rxjava/RequestContextAssemblyTest.java

```diff
@@ -0,0 +1,182 @@
+                          .subscribe((s, throwable) -> {
+                              res.complete(HttpResponse.of(HttpStatus.OK));
+                          });
+                    return HttpResponse.from(res);
+                }
+            });
+        }
+    };
+
+    private final ExecutorService pool = Executors.newCachedThreadPool();
+
+    Flowable<String> flowable(int count) {
+        RequestContext.current();
+        return Flowable.create(emitter -> {
+            pool.submit(() -> {
+                for (int i = 0; i < count; i++) {
+                    emitter.onNext("" + count);
+                }
+                emitter.onComplete();
+            });
+        }, BackpressureStrategy.BUFFER);
+    }
+
+    @Test
+    public void enableTracking() throws Exception {
+        try {
+            RequestContextAssembly.enable();
+            final HttpClient client = HttpClient.of(rule.uri("/"));
+            assertThat(client.execute(HttpHeaders.of(HttpMethod.GET, "/foo")).aggregate().get().status())
+                    .isEqualTo(HttpStatus.OK);
+        } finally {
+            RequestContextAssembly.disable();
+        }
+    }
+
+    @Test
+    public void withoutTracking() throws Exception {
+        final HttpClient client = HttpClient.of(rule.uri("/"));
+        assertThat(client.execute(HttpHeaders.of(HttpMethod.GET, "/foo")).aggregate().get().status())
+                .isEqualTo(HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+
+    @Test
+    public void composeWithOtherHook() throws Exception {
+        final AtomicInteger calledFlag = new AtomicInteger();
+        RxJavaPlugins.setOnSingleAssembly(single -> {
+            calledFlag.incrementAndGet();
+            return single;
+        });
+        final HttpClient client = HttpClient.of(rule.uri("/"));
+        client.execute(HttpHeaders.of(HttpMethod.GET, "/single")).aggregate().get();
+        // Single#create, #flatMap and #subscribe
+        assertThat(calledFlag.get()).isEqualTo(3);
+
+        try {
+            RequestContextAssembly.enable();
+            client.execute(HttpHeaders.of(HttpMethod.GET, "/single")).aggregate().get();
+            assertThat(calledFlag.get()).isEqualTo(6);
+        } finally {
+            RequestContextAssembly.disable();
+        }
+        client.execute(HttpHeaders.of(HttpMethod.GET, "/single")).aggregate().get();
+        assertThat(calledFlag.get()).isEqualTo(9);
+
+        RxJavaPlugins.setOnSingleAssembly(null);
+        client.execute(HttpHeaders.of(HttpMethod.GET, "/single")).aggregate().get();
+        assertThat(calledFlag.get()).isEqualTo(9);
+    }
+
+    private Single<String> single(String input) {
+        RequestContext.current();
+        return Single.create(emitter -> {
+            RequestContext.current();
+            emitter.onSuccess(input);
+        });
+    }
+
+    private Single<String> nestSingle(String input) {
+        RequestContext.current();
+        return single(input).flatMap(this::single);
```

Comment:
We can make all the methods as static so we can replace `RequestContextAssemblyTest.this::checkRequestContext` to `RequestContextAssemblyTest::checkRequestContext`

Reply:
`this::single` -> `RequestContextAssemblyTest::single`